v1/payroll/employee_time_worked/post.md

```diff
@@ -15,29 +15,75 @@ may be included for the employee.
+be hourly based (such as overtime premiums) or fixed amounts (such as tips or bonuses). The collection of `Additional Incomes`
+can contain hourly and fixed amounts though all items must be valid in order for the request to be accepted.
+
+If an entry for a time-based `CompanyAdditionalIncome` is submitted where `hours` are absent or is not greater than zero,
+the request will be rejected. The `amount` will be validated against the `rate` x `hours` x `additional income multiplier`.
+For example: An employee with an hourly rate of $25.00 per hour who did 2 hours of double overtime should the following in
+their `amount` field: `25.00 x 2 x 2 => 100.00`
+
+If an entry for a fixed-amount `CompanyAdditionalIncome` is submitted where `amount` is absent or is not greater than zero,
+the request will be rejected. The `hours` attribute must be `0`
+
+| Name   | Type      | Data Type                | Description                                                                                                        |
+| ------ | --------- | ------------------------ | ------------------------------------------------------------------------------------------------------------------ |
+| id     | required  | uuid (v4)                | UUID of the [CompanyAdditionalIncome](../../payroll/additional_incomes/get.md) this applies to                     |
+| hours  | required* | decimal formatted string | **Required on time-based additional incomes**. The number of hours the additional income applies for               |
+| amount | required* | money formatted string   | **Required on fixed-amount additional incomes**. The number of additional dollars earned during the period of work |
+
+
+**Inferred Additional Income**
+
+An `Inferred Additional Income` is provided for statutory holidays in which the hourly employee was provided a set wage. Inferred Additional Incomes
+will be applied to the employees primary rate that was active on the date of the statutory holiday.
+
+| Name       | Type     | Data Type              | Description                                                                                            |
+| ---------- | -------- | ---------------------- | ------------------------------------------------------------------------------------------------------ |
+| id         | required | uuid  (v4)             | UUID of the [CompanyAdditionalIncome](../../payroll/additional_incomes/get.md) (usually `holiday_pay`) |
+| start_time | required | datetime               | The start of the day for the statutory holiday (i.e. `2021-12-25 00:00:00 -05:00`                      |
+| end_time   | required | datetime               | The end of the day for the statutory holiday (i.e. `2021-12-25 23:59:59 -05:00`)                       |
+| amount     | required | money formatted string | The amount of money to be paid to the employee employee as compensation for the statutory holiday      |
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Right now accept hours if it's there, but calculate it from the amount if it's absent. Maybe we should add a section on calculating the hours from the amount (based on the rate)? It's somewhat not correct so the alternative would be to remove reading the hours all together from inferred_additional_income?

Reply:
@claus-humi do we accept hours right now? I thought we were always calculating it. If it's optional I can add that along with the caveat of how it works when it's absent.

Reply:
Looking at the code it seems like we do: https://github.com/Humi-HR/payroll/blob/master/app/services/time_worked_service/inferred_additional_income_creator_or_updater.rb#L46

Not sure if we should though?